include/nbl/config/BuildConfigOptions.h.in

```diff
@@ -106,6 +106,19 @@
     #define NBL_API2
 #endif
 
+// disabling 3rdparty warnings
+#if defined(_MVC_VER) && !defined(NBL_SILENT_WARNINGS_WITH_WRAPPER)
```

Comment:
I would make it

```c++
#if defined(NBL_SILENT_WARNINGS_WITH_WRAPPER)
  #if defined(_MVC_VER)
    #define NBL_PUSH_DISABLE_WARNINGS __pragma(warning(push, 0))
    #define NBL_POP_DISABLE_WARNINGS __pragma(warning(pop))
  // TODO: #elseif macro definitions for unix compilers
  #endif
#endif

#if !defined(NBL_PUSH_DISABLE_WARNINGS)
  #define NBL_PUSH_DISABLE_WARNINGS
#endif

#if !defined(NBL_POP_DISABLE_WARNINGS)
  #define NBL_POP_DISABLE_WARNINGS
#endif
```

Reply:
Personally, I prefer more compact solutions

Reply:
`#if defined(_MVC_VER) && !defined(NBL_SILENT_WARNINGS_WITH_WRAPPER)` makes your condition complicated, you will have a small trouble in `#else` statement then when you wanted to handle more toolsets